pyaedt/modeler/cad/Primitives.py

```diff
@@ -275,45 +587,4908 @@ def version(self):
+    ):
+        """Duplicate a selection around an axis.
+
+        Parameters
+        ----------
+        objid : list, str, int, Object3d or UserDefinedComponent
+            Name or ID of the object.
+        cs_axis :
+            Coordinate system axis or the Application.AXIS object.
+        angle : float, optional
+            Angle rotation in degees. The default is ``90``.
+        nclones : int, optional
+            Number of clones. The default is ``2``.
+        create_new_objects :
+            Whether to create the copies as new objects. The
+            default is ``True``.
+        is_3d_comp : bool, optional
+            If ``True``, the method will try to return the duplicated list of 3dcomponents. The default is ``False``.
+        duplicate_assignment : bool, optional
+            If True, the method duplicates selection assignments. The default value is ``True``.
+
+        Returns
+        -------
+        tuple
+
+        References
+        ----------
+
+        >>> oEditor.DuplicateAroundAxis
+        """
+        selections = self.convert_to_selections(objid)
+
+        vArg1 = ["NAME:Selections", "Selections:=", selections, "NewPartsModelFlag:=", "Model"]
+        vArg2 = [
+            "NAME:DuplicateAroundAxisParameters",
+            "CreateNewObjects:=",
+            create_new_objects,
+            "WhichAxis:=",
+            GeometryOperators.cs_axis_str(cs_axis),
+            "AngleStr:=",
+            self._arg_with_dim(angle, "deg"),
+            "Numclones:=",
+            str(nclones),
+        ]
+        vArg3 = ["NAME:Options", "DuplicateAssignments:=", duplicate_assignment]
+        added_objs = self.oeditor.DuplicateAroundAxis(vArg1, vArg2, vArg3)
+        self._duplicate_added_objects_tuple()
+        if is_3d_comp:
+            return self._duplicate_added_components_tuple()
+        return True, list(added_objs)
+
+    def _duplicate_added_objects_tuple(self):
+        added_objects = self.add_new_objects()
+        if added_objects:
+            return True, added_objects
+        else:
+            return False, []
+
+    def _duplicate_added_components_tuple(self):
+        added_component = self.add_new_user_defined_component()
+        if added_component:
+            return True, added_component
+        else:
+            return False, []
+
+    @pyaedt_function_handler()
+    def duplicate_along_line(
+        self,
+        objid,
+        vector,
+        nclones=2,
+        attachObject=False,
+        is_3d_comp=False,
+        duplicate_assignment=True,
+    ):
+        """Duplicate a selection along a line.
+
+        Parameters
+        ----------
+        objid : list, str, int, :class:`pyaedt.modeler.Object3d.Object3d`
+            Name or ID of the object.
+        vector : list
+            List of ``[x1,y1,z1]`` coordinates or the Application.Position object for
+            the vector.
+        attachObject : bool, optional
+            The default is ``False``.
```

Comment:
```suggestion
            List of ``[x1,y1,z1]`` coordinates or the ``Application.Position`` object for
            the vector.
        attachObject : bool, optional
            Whether to attach the object. The default is ``False``.
```
The description for ``attachObject`` likely needs to be improved.